package demo;

public class exceptions {
// stack overflow error
	
//	public static void m1() {
//		m2();
//	}
//	public static void m2(){
//		m1();
//	}
	static {
		int[] arr= new int[(Integer) null];
		System.out.println(arr.length);
		
	}
	
	public static void main(String[] args)   {
		
		
//		ClassCasrt Exception
		
		
//		Object s=new Object();
//	String o =(String)s;
//	System.out.println(o);
	
	
//		String s;
//		System.out.println(s.length());
		
		
//		Arrayindexoutofbounds exception
		
		
//		int[] a= new int[10];
//		a[0]=10;
//		a[9]=11;
//		try {
//		a[20]=13;
//		}
//		catch(ArrayIndexOutOfBoundsException e){
//			System.out.println(e.getMessage());
//		}System.out.println(a[0]);
//		System.out.println(a[2]);
//		
//	
//		customised exception
		
		
//		int age=19;
//		if(age<18) {
//			System.out.println("not eligible for voting");
//		throw new customException("non errors");
//		}
//		try {
//			
//			
//			System.out.println("try");
//		
//			 try {		
//				 System.out.println("inside nested try");
//				 System.out.println(10/0);
//			 }catch(ArithmeticException e) {
//				 System.out.println("inside nested catch");
//				 e.printStackTrace();
//			 }
//		}
//		catch(Exception e) {
//			System.out.println("catch");
//			e.printStackTrace();
//
//		}
//	
//		finally {
//			System.out.println("final");
//			
//		}
		
		
		
	}

}
